test-harness/modules/mocker/endpoints/recording.py

```diff
@@ -0,0 +1,137 @@
+        overridden in inheriting classes. Because of that all input arguments
+        are ignored.
+
+        Please refer to the description of the BaseHTTPRequestHandler class
+        for details on the arguments and return value of this method.
+        """
+        res = {"msg": 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'}
+        blob = self._convert_data_to_blob(res)
+
+        return blob
+
+    def _record_request(self):
+        """Store all the relevant data of the request into the endpoint context."""
+        ctx = self.server.context
+
+        res = {}
+        res['method'] = self.command
+        res['path'] = self.path
+        res['headers'] = self.headers.items()
+        res['request_version'] = self.request_version
+        if self.headers.get('Content-Length') is not None:
+            body_lenght = int(self.headers.get('Content-Length'))
+            res['request_body'] = self.rfile.read(body_lenght).decode('utf-8')
+        else:
+            res['request_body'] = None
+        res['request_time'] = time.time()
+
+        with ctx.lock:
+            ctx.data['requests'].append(res)
+        msg_fmt = "Endpoint `%s` recorded a request: `%s`"
+        log.debug(msg_fmt, ctx.data['endpoint_id'], res)
+
+    def _send_response(self, blob):
+        """Send response to the client.
+
+        The method handles:
+          * sending broken response if tests requested it
+          * recording the request data if enabled
+          * sending the default/good response in all other cases.
+
+        Please refer to the description of the BaseHTTPRequestHandler class
+        for details on the arguments of this method.
+        """
+        ctx = self.server.context
+        with ctx.lock:
+            # Seems a bit overkill:
+            do_record_request = ctx.data["record_requests"]
+            do_always_bork = ctx.data['always_bork']
+
+        if do_record_request:
+            self._record_request()
+
+        if do_always_bork:
+            msg_fmt = "Endpoint `%s` sending broken response as requested"
+            log.debug(msg_fmt, ctx.data['endpoint_id'])
+            blob = b"Broken response due to `always_bork` flag being set"
+            self._finalize_request(500, 'text/plain; charset=utf-8', blob)
+            return
+
+        self._finalize_request(200, 'application/json', blob)
+
+    @staticmethod
+    def _parse_request_body():
+        """Unused, just to satisfy the interface"""
+        return None
+
+
+# pylint: disable=C0103
+class RecordingTcpIpEndpoint(TcpIpHttpEndpoint):
+    """An endpoint that will record all the requests made to it.
+
+    This endpoint can be used to test features that work in the background
+    and are unavailable directly from the HTTP client context.
+
+    It's functionality is incomplete/serves only some example data, so most
+    probably it will have to be extended/inherited from in order to be able to
+    be able to serve as a mock.
```

Comment:
uhhh, yeah - silly :)

Changed to:
"In its current form, its functionality is incomplete/serves only some example data, it has to be extended/inherited from in order to serve as a mock."